packages/web/src/services/AudiusBackend.ts

```diff
@@ -246,123 +173,253 @@ export const fetchCID = async (
+    if (list) {
+      try {
+        return new Set(list.split(','))
+      } catch (e) {
+        console.error(e)
+        return null
+      }
     }
+  }
 
-    image.onerror = () => {
-      preloadImageTimer.end(start)
-      clearTimeout(timeout)
-      resolve(false)
+  function addDiscoveryProviderSelectionListener(
+    listener: DiscoveryProviderListener
+  ) {
+    didSelectDiscoveryProviderListeners.push(listener)
+    if (currentDiscoveryProvider !== null) {
+      listener(currentDiscoveryProvider)
     }
-    image.src = url
-  })
-}
+  }
 
-const fetchImageCID = async (
-  cid: CID,
-  creatorNodeGateways: string[] = [],
-  cache = true
-) => {
-  if (CIDCache.has(cid)) {
-    return CIDCache.get(cid)
+  function getCreatorNodeIPFSGateways(endpoint: Nullable<string>) {
+    if (endpoint) {
+      return endpoint
+        .split(',')
+        .filter(Boolean)
+        .map((endpoint) => `${endpoint}/ipfs/`)
+    }
+    const gateways = [`${userNodeUrl}/ipfs/`]
+    if (legacyUserNodeUrl) {
+      gateways.push(`${legacyUserNodeUrl}/ipfs/`)
+    }
+    return gateways
   }
 
-  creatorNodeGateways.push(`${USER_NODE}/ipfs`)
-  const primary = creatorNodeGateways[0]
-  if (primary) {
-    // Attempt to fetch/load the image using the first creator node gateway
-    const firstImageUrl = `${primary}${cid}`
-    const preloadedImageUrl = await preloadImage(firstImageUrl)
+  async function preloadImage(url: string) {
+    if (!preloadImageTimer) {
+      const batchSize =
+        getRemoteVar(IntKeys.IMAGE_QUICK_FETCH_PERFORMANCE_BATCH_SIZE) ??
+        undefined
 
-    // If the image is loaded, add to cache and return
-    if (preloadedImageUrl && cache) CIDCache.add(cid, preloadedImageUrl)
-    if (preloadedImageUrl) return preloadedImageUrl
-  }
+      preloadImageTimer = new Timer(
+        {
+          name: 'image_preload',
+          batch: true,
+          batchSize
+        },
+        ({ name, duration }) => {
+          console.info(`Recorded event ${name} with duration ${duration}`)
+          recordAnalytics(Name.PERFORMANCE, {
+            metric: name,
+            value: duration
+          })
+        }
+      )
+    }
 
-  await waitForLibsInit()
-  // Else, race fetching of the image from all gateways & return the image url blob
-  try {
-    const image = await audiusLibs.File.fetchCID(
-      cid,
-      creatorNodeGateways,
-      () => {}
-    )
+    return new Promise<string | false>((resolve) => {
+      const start = preloadImageTimer.start()
 
-    const url = NATIVE_MOBILE
-      ? image.config.url
-      : URL.createObjectURL(image.data)
+      const timeoutMs =
+        getRemoteVar(IntKeys.IMAGE_QUICK_FETCH_TIMEOUT_MS) ?? undefined
+      const timeout = setTimeout(() => {
+        preloadImageTimer.end(start)
+        resolve(false)
+      }, timeoutMs)
 
-    if (cache) CIDCache.add(cid, url)
+      // Avoid garbage collection by keeping a few images in an in-mem array
+      const image = new Image()
```

Comment:
this compat with native?

Reply:
Oooo I don't think so, good catch

Let me think about how to address this